crates/sui-sdk/examples/read_api.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+mod utils;
+
+use sui_sdk::rpc_types::{SuiGetPastObjectRequest, SuiObjectDataOptions};
+use sui_sdk::types::base_types::ObjectID;
+use sui_sdk::SuiClientBuilder;
+use utils::sui_address_for_examples;
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    let sui = SuiClientBuilder::default().build_testnet().await?; // testnet Sui network
+    println!("Sui testnet version{:?}", sui.api_version());
+    let active_address = sui_address_for_examples().await?; // get a random address for examples. Check utils module if you want to use a local wallet
+
+    // ************ READ API ************ //
+
+    println!("// ************ READ API ************ //\n");
+    // Owned Objects
+    let owned_objects = sui
+        .read_api()
+        .get_owned_objects(active_address, None, None, Some(5))
+        .await?;
+    println!(" *** Owned Objects ***");
+    println!("{:?}", owned_objects);
+    println!(" *** Owned Objects ***\n");
+
+    // Dynamic Fields
+    let dynamic_fields = sui
+        .read_api()
+        .get_dynamic_fields(ObjectID::from_address(active_address.into()), None, None)
+        .await?;
+    println!(" *** Dynamic Fields ***");
+    println!("{:?}", dynamic_fields);
+    println!(" *** Dynamic Fields ***\n");
+
+    let object = owned_objects
+        .data
+        .get(0)
+        .unwrap_or_else(|| panic!("No owned objects for this address {}", active_address));
+    let object_data = object
+        .data
+        .as_ref()
+        .unwrap_or_else(|| panic!("No object data for this SuiObjectResponse {:?}", object));
```

Comment:
Note that `clippy` will tell you to convert `.expect(format!(...))` into `.unwrap_or_else(|| panic!(...))` to avoid computing the formatted message unnecessarily.  Because this is an example, panicking is fine vs propagating the error, because when you propagate it, you lose the line position and backtrace.

Reply:
Indeed, clippy complained when I tried to change, so I left it as is.